src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java

```diff
@@ -199,4 +217,47 @@ private AdvertisementTitleDto buildAdvertisementTitle(Advertisement advertisemen
                 .ownerAvatar(advertisement.getUser().getAvatarImage())
                 .build();
     }
+
+    private AdvertisementDisplayDto buildAdvertisementDisplayDto(Advertisement advertisement) {
+        Location advLocation = advertisement.getLocation();
+        long id = advertisement.getId();
+        Subcategory subcategory = advertisement.getSubcategory();
+        String createdDate = advertisement.getCreated().format(DateTimeFormatter.ofPattern(dateFormat));
+        String age = Optional.ofNullable(advertisement.getAge()).map(AgeRange::getValue).orElse("");
+        return AdvertisementDisplayDto.builder()
+                .advertisementId(id)
+                .images(imageService.getByAdvertisementId(id))
+                .topic(advertisement.getTopic())
+                .description(advertisement.getDescription())
+                .location(convertTo(advLocation, LocationDto.class))
+                .ownerName(getOwnerFullName(advertisement))
+                .ownerAvatar(advertisement.getUser().getAvatarImage())
+                .age(age)
+                .dealType(advertisement.getDealType())
+                .season(advertisement.getSeason())
+                .size(advertisement.getSize())
+                .gender(advertisement.getGender())
+                .phone(getOwnerPhone(advertisement))
+                .readyForOffers(advertisement.isReadyForOffers())
+                .wishesToExchange(advertisement.getWishesToExchange())
+                .subcategory(convertTo(subcategory, SubcategoryDto.class))
+                .category(convertTo(subcategory.getCategory(), CategoryNameDto.class))
+                .createdDate(createdDate)
+                .build();
+    }
+
+    private String getOwnerPhone(Advertisement advertisement) {
+        return advertisement.getUser().getPhones().stream()
+                .findFirst()
+                .map(phone -> String.valueOf(phone.getPhoneNumber()))
+                .orElse("");
+    }
+
+    private String getOwnerFullName(Advertisement advertisement) {
+        User owner = advertisement.getUser();
+        String firstName = Optional.of(owner.getFirstName()).orElse("");
+        String lastName = Optional.of(owner.getLastName()).orElse("");
```

Comment:
would you like to put a space here?
String ownerName = String.format("%s%s%s", firstName()," " ,lastName()).trim();

Reply:
I have added a space in a different way.
Why your approach is better?

Reply:
String.format("%s %s", firstName(), "lastName()).trim();
We already have space here between first name and last name.
You can view it in the test case.

Reply:
not only did not see, but was stupid

Reply:
Guys, why so complicated???
```
private String getOwnerFullName(User user) {
        String formatted = String.format("%s %s", user.getFirstName(), user.getLastName());
        return formatted.isBlank() ? user.getUsername() : formatted;
    }
```

Reply:
Thank you. Changed.